pkg/config/controller.go

```diff
@@ -46,4 +46,9 @@ type Controller struct {
 	// NeedLeaderElection indicates whether the controller needs to use leader election.
 	// Defaults to true, which means the controller will use leader election.
 	NeedLeaderElection *bool
+
+	// WatchProviderClusters indicates whether the controller should
+	// only watch clusters that are engaged by the cluster provider. Defaults to false
+	// if no provider is set, and to true if a provider is set.
+	WatchProviderClusters *bool
```

Comment:
Should this be named WatchOnlyProviderClusters? (currently it sounds like we watch provider clusters in addition to the hub cluster)

Reply:
I guess that also wouldn't be better because false would signal that we're watching both the provider and hub clusters

Reply:
Updated the PR. This is not relevant anymore.

Compare in controller options:
```golang
	// EngageWithDefaultCluster indicates whether the controller should engage
	// with the default cluster of a manager. This defaults to false through the
	// global controller options of the manager if a cluster provider is set,
	// and to true otherwise. Here it can be overridden.
	EngageWithDefaultCluster *bool
	// EngageWithProvidedClusters indicates whether the controller should engage
	// with the provided clusters of a manager. This defaults to true through the
	// global controller options of the manager if a cluster provider is set,
	// and to false otherwise. Here it can be overridden.
	EngageWithProviderClusters *bool
```

Reply:
That's way clearer, thx!